hash-tables/problem-set-hash-tables.md

```diff
@@ -0,0 +1,287 @@
+# Problem Set: Hash Table
+
+## Directions
+
+Complete all questions below. **Solve all questions using a hash table** (Python dictionary).
+
+## Practice
+
+<!-- Question 1 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: 9U2j0J
+* title: Hash Tables
+* points: 3
+### !question
+
+Write a function `get_intersection` that takes two integer lists with unique values and returns their intersection in a new array.
+
+Be sure to use a dictionary in your solution. Do not use any built-in intersection functions.
+
+Here are the tests:
+
+```python
+def test_get_intersection_one_match():
+    red = [2, 3, 4]
+    blue = [4, 5, 6]
+    assert get_intersection(red, blue) == [4]
+
+
+def test_get_intersection_no_matches():
+    red = [9, 30, 42]
+    blue = [56, 34, 90, 32]
+    assert get_intersection(red, blue) == []
+
+
+def test_get_intersection_many_matches():
+    red = [50, 43, 25, 72]
+    blue = [25, 36, 43, 50, 80]
+    result = get_intersection(red, blue)
+    assert 25 in result
+    assert 43 in result
+    assert 50 in result
+    assert len(result) == 3
+```
+
+### !end-question
+### !tests
+```python
+import unittest
+from main import *
+class TestChallenge(unittest.TestCase):
+    # Note: Written with these asserts to mimic the pytest assertions we give students
+
+    def test_get_intersection_one_match(self):
+        red = [2, 3, 4]
+        blue = [4, 5, 6]
+        self.assertEqual(get_intersection(red, blue), [4])
+
+    def test_get_intersection_no_matches(self):
+        red = [9, 30, 42]
+        blue = [56, 34, 90, 32]
+        self.assertEqual(get_intersection(red, blue), [])
+
+    def test_get_intersection_many_matches(self):
+        red = [50, 43, 25, 72]
+        blue = [25, 36, 43, 50, 80]
+        result = get_intersection(red, blue)
+        self.assertTrue(25 in result)
+        self.assertTrue(43 in result)
+        self.assertTrue(50 in result)
+        self.assertEqual(len(result), 3)
+```
+### !end-tests
+### !explanation
+
+An example of a working implementation:
+
+```python
+def get_intersection(red, blue):
+    frequency_hash = {}
+
+    for list in [red, blue]:
+        for item in list:
+            if frequency_hash.get(item):
```

Comment:
for the puer boolean approach here, maybe show using `in` instead (setting the standard of depending on the `get` result without a discussion of values that could be misinterpreted as a missing key) makes me a little uncomfortable

```suggestion
            if item in frequency_hash:
```

But we could avoid the conditional completely by using the version of get that takes a default. If students haven't seen this version yet, they should feel comfortable looking it up
```python
            count = frequency_hash.get(item, 0)
            frequency_hash[item] = count + 1
```